modules/SofaBoundaryCondition/ConstantForceField.inl

```diff
@@ -45,165 +48,358 @@ namespace forcefield
-{
-    d_arrowSizeCoef.setGroup("Visualization");
-    d_color.setGroup("Visualization");
+    if (d_indices.isSet() && indicesSize!=0)
+    {
+        // check size of vector indices
+        if( indicesSize > m_systemSize )
+        {
+            msg_error() << "Size mismatch: indices > system size";
+            this->m_componentstate = core::objectmodel::ComponentState::Invalid;
+        }
+        // check each indice of the vector
+        for(size_t i=0; i<indicesSize; i++)
+        {
+            if( indices[i] > m_systemSize )
+            {
+                msg_error() << "Indices incorrect: indice["<< i <<"] = "<< indices[i] <<" exceeds system size";
+                this->m_componentstate = core::objectmodel::ComponentState::Invalid;
+            }
+        }
+    }
+    else
+    {
+        // initialize with all indices
+        VecIndex& indicesEdit = *d_indices.beginEdit();
+        indicesEdit.clear();
+        indicesEdit.resize(m_systemSize);
+        std::iota (std::begin(indicesEdit), std::end(indicesEdit), 0);
+        d_indices.endEdit();
+    }
 
-    /// We add aliases only for deprecated attributes and as long as we are handling them.
-    addAlias(&d_indices, "points") ;
+    if (d_forces.isSet())
+    {
+        const VecDeriv &forces = d_forces.getValue();
+        if( checkForces(forces) )
+        {
+            computeForceFromForceVector(forces);
+        }
+        else
+        {
+            msg_error() << " Invalid given vector forces";
+            this->m_componentstate = core::objectmodel::ComponentState::Invalid;
+        }
+        msg_info() << "Input vector forces is used for initialization";
+    }
+    else if (d_force.isSet())
+    {
+        const Deriv &force = d_force.getValue();
+        if( checkForce(force) )
+        {
+            computeForceFromSingleForce(force);
+        }
+        else
+        {
+            msg_error() << " Invalid given force";
+            this->m_componentstate = core::objectmodel::ComponentState::Invalid;
+        }
+        msg_info() << "Input force is used for initialization";
+    }
+    else if (d_totalForce.isSet())
+    {
+        const Deriv &totalForce = d_totalForce.getValue();
+        if( checkForce(totalForce) )
+        {
+            computeForceFromTotalForce(totalForce);
+        }
+        else
+        {
+            msg_error() << " Invalid given totalForce";
+            this->m_componentstate = core::objectmodel::ComponentState::Invalid;
+        }
+        msg_info() << "Input totalForce is used for initialization";
+    }
+
+    // init from ForceField
+    Inherit::init();
+
+    // add to tracker
+    this->trackInternalData(d_indices);
+    this->trackInternalData(d_forces);
+    this->trackInternalData(d_force);
+    this->trackInternalData(d_totalForce);
+
+    // if all init passes, component is valid
+    this->m_componentstate = core::objectmodel::ComponentState::Valid;
```

Comment:
This will always be true so the component will always be set to Valid.
You should replace all the lines: ```this->m_componentstate = core::objectmodel::ComponentState::Invalid;``` by a bool ready = false;

and here 
```
if (ready) 
  {this->m_componentstate = core::objectmodel::ComponentState::Valid;}
else
   {this->m_componentstate = core::objectmodel::ComponentState::Invalid;}

Reply:
oh I didn't see the return, sorry forgot my remark.

Reply:
ok you added it. I'm not so crazy. Yes return is even better because you won't have Data tracked. Less chance to compute things in an invalid state.